Sources/WebAuthenticationViewController.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  WebAuthenticationViewController.swift
+//  SwifteriOS
+//
+//  Created by Huiping Guo on 2020/01/14.
+//  Copyright © 2020 Matt Donnelly. All rights reserved.
+//
+
+import Foundation
+import UIKit
+import SafariServices
+
+class WebAuthenticationViewController: UIViewController {
+    
+    let queryUrl: URL
+    
+    let dismissHandler: (() -> Void)?
+    
+    init(queryUrl: URL, dismissHandler: (() -> Void)? = nil) {
+        self.queryUrl = queryUrl
+        self.dismissHandler = dismissHandler
+        super.init(nibName: nil, bundle: nil)
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+                
+        embedViewController(viewController: createWebViewController())
+    }
+    
+    func embedViewController(viewController: UIViewController) {
```

Comment:
Not a fan of this at all. We shouldn't embed the view controller just to conform to `SFSafariViewControllerDelegate`. I propose having a default concrete object that conforms to the delegate that is owned by the `Swifter` object (that gets `nil`'d out when it's done) in the event the client provides a `nil` delegate. Either that, or the `Swifter` class itself can just conform to the SFSafariViewControllerDelegate that has a default behaviour of dismissing it on completion.